src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -470,5 +509,45 @@ bool DefaultOTARequestorDriver::GetNextProviderLocation(ProviderLocationType & p
     return false;
 }
 
+CHIP_ERROR DefaultOTARequestorDriver::ScheduleRetry(bool trySameProvider)
+{
+    VerifyOrDie(mRequestor != nullptr);
+
+    ProviderLocationType providerLocation;
+    CHIP_ERROR status        = CHIP_NO_ERROR;
+    bool willTryAnotherQuery = true;
+
+    if (trySameProvider == false)
+    {
+        bool listExhausted = false;
+        if ((GetNextProviderLocation(providerLocation, listExhausted) != true) || (listExhausted == true))
+        {
+            willTryAnotherQuery = false;
```

Comment:
Why the need for this variable? Why not just return early (with the appropriate CHIP_ERROR) if we're not going to schedule a query?

Reply:
This code came from `UpdateNotFound()` mostly as-is.  Let me see if I can get rid of `willTryAnotherQuery`.